server/src/lib.rs

```diff
@@ -67,10 +71,49 @@ pub async fn run(
     address: String,
     database_path: String,
     storage_path: String,
+    local_distributed_address: String,
+    all_servers_address: Vec<String>,
 ) -> anyhow::Result<()> {
     let listener = TcpListener::bind(&address).await?;
     let local_storage = Arc::new(DefaultEngine::new(&database_path, &storage_path));
-    let engine = Arc::new(DistributedEngine::new(address.clone(), local_storage));
+    local_storage.init();
+    build_hash_ring(all_servers_address.clone());
+    let service = RPCService::new(local_storage.clone());
+    let local = local_distributed_address.clone();
+    tokio::spawn(async move {
+        tonic::transport::Server::builder()
+            .add_service(engine_rpc::new_manager_service(service))
+            .serve((&local).parse().unwrap())
+            .await
+    });
+    let engine = Arc::new(DistributedEngine::new(
+        local_distributed_address.clone(),
+        local_storage.clone(),
+    ));
+    for value in all_servers_address.clone().into_iter() {
+        engine.add_connection(value, None);
+    }
+    for value in all_servers_address.clone().into_iter() {
+        if local_distributed_address == value.clone() {
+            continue;
+        }
+        let server_address = format!("http://{}", value);
+        let arc_engine = engine.clone();
+        tokio::spawn(async move {
+            let mut interval = time::interval(time::Duration::from_secs(5));
```

Comment:
5 seconds might be too long

Reply:
I think it's better to always try, because other servers may go offline unexpectedly.

Reply:
Changed to 1 second